AR-Top-MobileApp/Assets/Prefabs/JSONReader.cs

```diff
@@ -0,0 +1,239 @@
+		mapScaler = GameObject.Find ("MapScaler");
+		if (mapScaler == null) {
+			mapScaler = new GameObject ("MapScaler");
+		}
+
+		//Create an empty parent for the grid.
+		gridLayout = GameObject.Find ("GridLayout");
+		if (gridLayout == null){
+			gridLayout = new GameObject("GridLayout");
+			gridLayout.transform.SetParent (mapScaler.transform);
+		}
+
+		//Create an empty parent for the Player Character pieces.
+		playerLayout = GameObject.Find ("PlayerLayout");
+		if (playerLayout == null) {
+			playerLayout = new GameObject ("PlayerLayout");
+			playerLayout.transform.SetParent (mapScaler.transform);
+		}
+
+		//Create an empty parent for the NPC pieces.
+		npcLayout = GameObject.Find ("NPCLayout");
+		if (npcLayout == null) {
+			npcLayout = new GameObject ("NPCLayout");
+			npcLayout.transform.SetParent (mapScaler.transform);
+		}
+
+		//Create an empty parent for the Tile pieces.
+		tileLayout = GameObject.Find ("TileLayout");
+		if (tileLayout == null) {
+			tileLayout = new GameObject ("TileLayout");
+			tileLayout.transform.SetParent (mapScaler.transform);
+		}
+
+		//This creates a game object to play around with. Eventually this
+		//will be replaced by retrieving the JSON object from the server.
+		GridItem obj = new GridItem ();
+		obj.type = "character";
+		obj.x = 1;
+		obj.y = 0;
+		obj.z = 1;
+		obj.red = 0.2f;
+		obj.green = 1f;
+		obj.blue = 0.2f;
+		obj.alpha = 1f;
+
+		GridItem grid = new GridItem ();
+		grid.type = "grid";
+		grid.x = 5;
+		grid.y = 2;
+		grid.z = 7;
+		grid.red = 0.2f;
+		grid.green = 0.2f;
+		grid.blue = 0.2f;
+		grid.alpha = 1f;
+
+		GridItem tilepiece = new GridItem ();
+		tilepiece.type = "tile";
+		tilepiece.x = 3;
+		tilepiece.y = 0;
+		tilepiece.z = 3;
+		tilepiece.red = 1f;
+		tilepiece.green = 0f;
+		tilepiece.blue = 0f;
+		tilepiece.alpha = 1f;
+
+		GridItem tilepiece1 = new GridItem ();
+		tilepiece1.type = "tile";
+		tilepiece1.x = 4;
+		tilepiece1.y = 0;
+		tilepiece1.z = 3;
+		tilepiece1.red = 1f;
+		tilepiece1.green = 0f;
+		tilepiece1.blue = 0f;
+		tilepiece1.alpha = 1f;
+
+		GridItem tilepiece2 = new GridItem ();
+		tilepiece2.type = "tile";
+		tilepiece2.x = 3;
+		tilepiece2.y = 0;
+		tilepiece2.z = 4;
+		tilepiece2.red = 1f;
+		tilepiece2.green = 0f;
+		tilepiece2.blue = 0f;
+		tilepiece2.alpha = 1f;
+
+		GridItem tilepiece3 = new GridItem ();
+		tilepiece3.type = "tile";
+		tilepiece3.x = 4;
+		tilepiece3.y = 0;
+		tilepiece3.z = 4;
+		tilepiece3.red = 1f;
+		tilepiece3.green = 0f;
+		tilepiece3.blue = 0f;
+		tilepiece3.alpha = 1f;
+
+		GridItem enemy = new GridItem ();
+		enemy.type = "npc";
+		enemy.x = 4;
+		enemy.y = 0;
+		enemy.z = 3;
+		enemy.red = 0f;
+		enemy.green = 0f;
+		enemy.blue = 0f;
+		enemy.alpha = 1f;
+
+		//builds the piece.
+		buildPiece (obj);
+		buildPiece (grid);
+		buildPiece (tilepiece);
+		buildPiece (tilepiece1);
+		buildPiece (tilepiece2);
+		buildPiece (tilepiece3);
+		buildPiece (enemy);
+	}
+	
+	// Update is called once per frame
```

Comment:
Can we remove this for now, since we're not using it?

Reply:
We will likely need this eventually (in a sprint or two). It's a function that Unity adds whenever a new script is made. It is basically called at the start of each frame and performs whatever is in it. We CAN remove it and add it later but it's not hurting anything at this time.